src/main/java/SoftDeleteSample.java

```diff
@@ -156,38 +178,45 @@ public static void deletedKeyRecovery() throws Exception {
         String keyToRecoverName = getRandomName("key");
         String keyToPurgeName = getRandomName("key");
 
-        KeyBundle key = keyVaultClient.createKey(vault.vaultUri(), keyToRecoverName, JsonWebKeyType.RSA);
-        System.out.printf("Create key %s.%n", key.keyIdentifier().toString());
 
-        key = keyVaultClient.createKey(vault.vaultUri(), keyToPurgeName, JsonWebKeyType.RSA);
-        System.out.printf("Create key %s.%n", key.keyIdentifier().toString());
+        KeyAsyncClient keyAsyncClient = vault.keyClient();

```

Comment:
I understand the `vault.keyClient()` method returns a `KeyAsyncClient`, so this kind of forces us to block if we want to make the code synchronous so it's easier to read for people not familiar with chaining reactive operations.

Have we considered directly creating a sync `KeyClient` using the same credentials and vault URL from the vault object resulting from creating a key vault?

```suggestion
        KeyClient keyClient = new KeyClientBuilder()
            .credential(createToken())
            .vaultUrl(vault.vaultUri())
            .buildClient();
```